xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1097,3 +1130,124 @@ void DX::DeviceResources::Trim() const
 }
 
 #endif
+
+void DX::DeviceResources::DetectDisplayHDRcapable(bool& hdr_capable, bool& hdr_enabled) const
+{
+  ComPtr<IDXGIOutput> pOutput;
+  ComPtr<IDXGIOutput6> pOutput6;
+  DXGI_OUTPUT_DESC1 outDesc1;
+  hdr_capable = false;
+  hdr_enabled = false;
+
+  if (m_swapChain == nullptr)
+    return;
+
+  if (SUCCEEDED(m_swapChain->GetContainingOutput(pOutput.GetAddressOf())))
+  {
+    if (SUCCEEDED(pOutput.As(&pOutput6)))
+    {
+      if (SUCCEEDED(pOutput6->GetDesc1(&outDesc1)))
+      {
+        if (outDesc1.ColorSpace == DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020)
+        {
+          CLog::LogF(LOGNOTICE, "Monitor HDR capable detected and HRD current state is ENABLED:");
+          hdr_enabled = true;
+          hdr_capable = true;
+        }
+        else if (outDesc1.MaxLuminance >= 400.0)
+        {
+          CLog::LogF(LOGNOTICE, "Monitor HDR capable is detected but NOT enabled:");
+          hdr_capable = true;
+        }
+        else
+        {
+          CLog::LogF(LOGNOTICE, "No monitor HDR capable detected.");
+        }
+        if (hdr_capable)
+        {
+          CLog::LogF(LOGNOTICE, " - BitsPerColor = %d", outDesc1.BitsPerColor);
+          CLog::LogF(LOGNOTICE, " - ColorSpace = %d", outDesc1.ColorSpace);
+          CLog::LogF(LOGNOTICE, " - RedPrimary = %0.3f, %0.3f", outDesc1.RedPrimary[0], outDesc1.RedPrimary[1]);
+          CLog::LogF(LOGNOTICE, " - GreenPrimary = %0.3f, %0.3f", outDesc1.GreenPrimary[0], outDesc1.GreenPrimary[1]);
+          CLog::LogF(LOGNOTICE, " - BluePrimary = %0.3f, %0.3f", outDesc1.BluePrimary[0], outDesc1.BluePrimary[1]);
+          CLog::LogF(LOGNOTICE, " - WhitePoint = %0.3f, %0.3f", outDesc1.WhitePoint[0], outDesc1.WhitePoint[1]);
+          CLog::LogF(LOGNOTICE, " - MinLuminance = %0.3f", outDesc1.MinLuminance);
+          CLog::LogF(LOGNOTICE, " - MaxLuminance  = %0.0f", outDesc1.MaxLuminance);
+          CLog::LogF(LOGNOTICE, " - MaxFullFrameLuminance = %0.0f", outDesc1.MaxFullFrameLuminance);
+        }
+      }
+      else
+      {
+        CLog::LogF(LOGERROR, "IDXGIOutput6 -> GetDesc1 FAILED");
+      }
+    }
+  }
+}
+
+void DX::DeviceResources::Set_HDR_MetaData(AVMasteringDisplayMetadata dm, AVContentLightMetadata lm) const
```

Comment:
please don't use ffmpeg structs in DX::DeviceResources. pass prepared struc `DXGI_HDR_METADATA_HDR10` from video renderer.

Reply:
Done!